### PR TITLE
Update ethers rs version

### PR DESCRIPTION
**Pull request by sree-revoori1**
It is currently very hard to use this sdk and ethers rs in tandem due to the version of ethers being very outdated and types mismatching. This PR updates the ethers version to the latest version.

Also, run cargo fmt, run cargo clippy, and ignore Cargo.lock as this is a library.

### CODE REVIEW

File: src/signature/create_signature.rs [hunk 1]
```diff
@@ -93,21 +98,23 @@ fn sign_typed_data<T: Eip712>(payload: &T, wallet: &LocalWallet) -> Result<Signa
...
 
-    Ok(sign_hash(H256::from(encoded), wallet))
+    sign_hash(H256::from(encoded), wallet)
 }
 
-fn sign_hash(hash: H256, wallet: &LocalWallet) -> Signature {
-    let recoverable_sig: recoverable::Signature =
-        wallet.signer().sign_digest(Sha256Proxy::from(hash));
+fn sign_hash(hash: H256, wallet: &LocalWallet) -> Result<Signature> {
+    let (sig, recid) = wallet
```
Comment (lmlmt) [inline src/signature/create_signature.rs:105]: nit: `rec_id`?

**Review by lmlmt — APPROVED** [review]
Looks good, will merge once you fix merge conflicts

File: src/signature/create_signature.rs (hunk 1 above)
Comment (sree-revoori1) [reply src/signature/create_signature.rs]: Done

**Comment by sree-revoori1** [comment]
@lmlmt resolved the merge conflicts.

**Pull request merged**